File: matlab/plot_SIM.m
% Plot MSE versus the transmit power
% L = 64; N = 4; K = 10; nbrBSs = 4; nbrgroups = 1; tau_p = 10; p_pilot = 100 mW;

% Distribution 1: users in group g uniformly ditrsiued in cell g

%Uplink transmit power per UE (dBm)
p = [2,3,4,5,6];

% FlatCE_Net = [0.284101, 0.24624, 0.194411, 0.216062, 0.206377];
% LS_NMSE = [11.096421, 3.560983, 1.690374, 1.9375, 1.768164];
% MMSE_NMSE = [0.242319863, 0.220775224, 0.173299815, 0.186430324, 0.190794827];
% CDRN_NMSE = [0.216069, 0.187047, 0.06501, 0.04531, 0.016515];

FlatCE_Net = [0.284101, 0.24624, 0.194411, 0.216062, 0.206377];
LS_NMSE = [11.096421, 3.560983, 1.690374, 1.9375, 1.768164];
MMSE_NMSE = [0.242319863, 0.220775224, 0.173299815, 0.186430324, 0.190794827];
CDRN_NMSE = [0.32681,0.280037,0.216069,0.237657,0.245519];


fig = figure('Visible', 'off');
hold on; box on;
% plot(p, FlatCE_Net, 'b->','LineWidth',1.3, 'MarkerSize',8);
% plot(p, LS_NMSE, 'k-^','LineWidth',1.3, 'MarkerSize',8);
% plot(p, MMSE_NMSE, 'k--^','LineWidth',1.3, 'MarkerSize',8);
% plot(p, CDRN_NMSE, 'm-s','LineWidth',1.3, 'MarkerSize',8);

% 1) FlatCE_Net：实线 + 红色 + 圆点标记
plot(p, FlatCE_Net, 'r-o','LineWidth',1.5, 'MarkerSize',8);

% 2) LS_NMSE：虚线 + 蓝色 + 三角标记
plot(p, LS_NMSE, 'b--^','LineWidth',1.5, 'MarkerSize',8);

% 3) MMSE_NMSE：虚线 + 黑色 + 方块标记（最优解）
plot(p, MMSE_NMSE, 'k--s','LineWidth',1.5, 'MarkerSize',8);

% 4) CDRN_NMSE：虚线 + 洋红色 + 菱形标记
plot(p, CDRN_NMSE, 'm--d','LineWidth',1.5, 'MarkerSize',8);

% plot(p, averageMSE_Level3_with_TCO, 'm--s','LineWidth',1.3, 'MarkerSize',8);

xlabel('Number of layers','Interpreter','Latex','Fontname', 'Times New Roman');
ylabel('Normed MSE','Interpreter','Latex','Fontname', 'Times New Roman');

legend({'FlatCE\_Net','LS\_NMSE','MMSE\_NMSE','CDRN\_NMSE'},...
       'Interpreter','Latex','Fontname', 'Times New Roman', 'Location', 'Best');


set(gca,'FontSize',18);
set(gca, 'LooseInset', [0,0,0,0]);
set(gca, 'YScale', 'log');


% 保存图像到文件 (可以选择不同格式，如 PNG、EPS、PDF)
saveas(fig, '/users/elq20xd/workspace/channel_estimation/flatCE_Net/figures/SIM_performance_comparison_120225_log.png'); % 保存为 PNG 格式
% saveas(fig, '/users/elq20xd/workspace/channel_estimation/flatCE_Net/figures/SIM_performance_comparison_120225_log.pdf'); % 保存为 PDF 格式
saveas(fig, '/users/elq20xd/workspace/channel_estimation/flatCE_Net/figures/SIM_performance_comparison_120225_log.eps', 'epsc'); % 保存为 EPS 格式
% print(fig, 'MSE_vs_TransmitPower', '-dpng', '-r300'); % 高分辨率保存 PNG

% 关闭图像窗口，避免占用内存
close(fig);

File: utils/plot_SIM.py
import matplotlib.pyplot as plt
import os

# 数据准备
L = [2,3,4,5,6] # dBm 横轴数据
# SDUNet_64 = [0.398879,0.143324,0.098255, 0.068289, 0.065288] 
# LS_NMSE = [28.059807, 1.239765, 0.38713, 0.199826, 0.196495]
# MMSE_NMSE = [0.12830184, 0.074577348, 0.051158786, 0.041384909, 0.040207503]

FlatCE_Net = [0.284101, 0.24624, 0.194411, 0.216062, 0.206377]
LS_NMSE = [11.096421, 3.560983, 1.690374, 1.9375, 1.768164]
MMSE_NMSE = [0.242319863, 0.220775224, 0.173299815, 0.186430324, 0.190794827]


# SDUNet_64 = [0.143324,0.098255, 0.068289, 0.065288] 
# LS_NMSE = [1.239765, 0.38713, 0.199826, 0.196495]
# MMSE_NMSE = [0.074577348, 0.051158786, 0.041384909, 0.040207503]

# 绘图
plt.figure(figsize=(8, 6))
# plt.plot(L, SDUNet_64, 's-', label='SDUNet_64', linewidth=1.5, color='#0072BD')
# plt.plot(L, LS_NMSE, 'd-', label='LS method', linewidth=1.5, color='#77AC30')
# plt.plot(L, MMSE_NMSE, 'p:', label='MMSE method', linewidth=1.5, color='#A2142F')
# plt.plot(L, CDRN_NMSE, 'd-.', label='SIM-Optimized, N=64', linewidth=1.5, color='#77AC30')
plt.plot(L, FlatCE_Net, 's-', label='FlatCE-Net', linewidth=2, color='#1f77b4', markersize=8)  # 蓝色, 方形标记
plt.plot(L, LS_NMSE, 'p:', label='LS method', linewidth=2, color='#ff7f0e', markersize=8)   # 橙色, 圆形标记
plt.plot(L, MMSE_NMSE, '^:', label='MMSE method', linewidth=2, color='#2ca02c', markersize=8)  # 绿色, 三角形标记
# plt.plot(L, CDRN_NMSE, 'd:', label='CDRN method', linewidth=2, color='pink', markersize=8)  # 红色, 叉号标记



# 图例
plt.legend(loc='upper right', fontsize=10)

# plt.yscale('log')  # y 轴取对数
plt.xlabel('L', fontsize=12)
plt.ylabel('Normed MSE (log scale)', fontsize=12)
plt.title('Performance Comparison with Logarithmic Y-Axis', fontsize=14)
plt.grid(True, which='both', linestyle='--', linewidth=0.5, alpha=0.7)

plt.xticks(range(2, 7, 1))

# 额外的文本标注
# plt.text(5.5, 0.4, 'Solid Lines: N = 32, M = 4', fontsize=10)
# plt.text(5.5, 0.35, 'Dashdotted Lines: N = 64, M = 4', fontsize=10)

# 保存路径
output_folder = '/users/elq20xd/workspace/channel_estimation/flatCE_Net/figures'
os.makedirs(output_folder, exist_ok=True)

# 保存为 PNG 和 EPS
plt.tight_layout()
plt.savefig(os.path.join(output_folder, 'FlatCENet_SIM_performance_comparison_nolog.png'), dpi=300)
# plt.savefig(os.path.join(output_folder, 'performance_comparison.eps'), format='eps')
plt.savefig(os.path.join(output_folder, 'FlatCENet_SIM_performance_comparison_nolog.pdf'), format='pdf')


# 显示图形
# plt.show()
